### PR TITLE
add task7

### PR DESCRIPTION
**Pull request by vkhmara**

### CODE REVIEW

File: task7/Working with posts.js [hunk 1]
```diff
@@ -0,0 +1,353 @@
+(function(){
+class Post {
```
Comment (skateray17) [inline task7/Working with posts.js:2]: Can I somehow like/dislike post?

File: task7/Working with posts.js [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+
+	_isString(obj) {
+		return typeof obj == 'string';
+	}
+
+	get isString() {
+		return this._isString;
+	}
+
+  _rightDate(date) {
```
Comment (skateray17) [inline task7/Working with posts.js]: Wrong indentation in this function

File: task7/Working with posts.js [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+
+	_getPostsByFilter(filterConfig) {
+		return this.posts.filter(element=>
+			(filterConfig.author == "" || element.author == filterConfig.author) &&
+			((filterConfig.dateInterval == null) || this.liesInInterval(element.createdAt, filterConfig.dateInterval)) &&
+			 ((filterConfig.hashTags == []) || this.hasAnyTag(element.hashTags, filterConfig.hashTags))
+			 );
+	}
+
+	get getPostsByFilter() {
```
Comment (skateray17) [inline task7/Working with posts.js]: Why is this getter needed?

File: task7/Working with posts.js [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+		this.count += 1;
+		return true;
+	}
+
+	editPost(ID, edit) {
+		let index = this.posts.findIndex(item=>item.id==ID);
+		if (index==-1) {
+			return false;
+		}
+		let edPost = this.posts[index];
```
Comment (skateray17) [inline task7/Working with posts.js]: Object is reference type changing it here changes it in array. So, validation can be failed, but post already changed

File: task7/Working with posts.js [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+			return false;
+		}
+		this.posts.splice(index, 1);
+		this.count -= 1;
+		return true;
+	}
+
+	addAll(arrayPosts) {
+		let invalidPosts = [];
+		arrayPosts.forEach(item => {
```
Comment (skateray17) [inline task7/Working with posts.js]: Can you try to use `reduce` method for it? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce

File: task7/Working with posts.js [hunk 6]
```diff
@@ -0,0 +1,353 @@
...
+
+function printLengthOfArray(arr) {
+	document.write('The length of posts ' + arr.length + '<br\/>');
+}
+
+class PostContainer {
+
+	constructor(posts) {
+		this._posts = posts;
+		this._count = this._posts.length;
```
Comment (skateray17) [inline task7/Working with posts.js]: _count is redundant since it should always be equal to _posts.length

**Comment by vkhmara** [comment]
I've done

**Review by skateray17 — APPROVED** [review]

**Pull request merged**